ui/components/component-library/tag/tag.types.ts

```diff
@@ -0,0 +1,27 @@
+import { TextProps } from '../text/text.types';
+import type {
+  StyleUtilityProps,
+  PolymorphicComponentPropWithRef,
+} from '../box';
+
+export interface TagStyleUtilityProps extends StyleUtilityProps {
+  /**
+   * The text content of the Tag component, can either be a string or ReactNode
+   */
+  label?: string | React.ReactNode;
+  /**
+   * The label props of the component. Most Text component props can be used
+   */
+  labelProps?: TextProps<'span'>;
```

Comment:
I suppose this is fine but would note that this prevents things like `<Tag labelProps={{ as: 'p' }} />`